contracts/OlasMech.sol

```diff
@@ -90,118 +87,91 @@ abstract contract OlasMech is Mech, IErrorsMech, ImmutableStorage {
+            // Previous element of the current next element will be the newly created element
+            mapRequestIds[curNextRequestIdLink][0] = requestId;
+
+            emit Request(requester, requestId, datas[i]);
+        }
 
         // Increase the number of undelivered requests
-        numUndeliveredRequests++;
+        numUndeliveredRequests += numRequests;
         // Increase the total number of requests
-        numTotalRequests++;
-
-        emit Request(requester, requestId, data);
+        numTotalRequests += numRequests;
     }
 
     /// @dev Cleans the request info from all the relevant storage.
-    /// @param requester Requester address.
     /// @param requestId Request Id.
-    function _cleanRequestInfo(address requester, uint256 requestId) internal virtual {
-        // Decrease the number of undelivered requests
-        mapUndeliveredRequestsCounts[requester]--;
-        numUndeliveredRequests--;
-
-        // Remove delivered request Id from the request Ids map
-        uint256[2] memory requestIds = mapRequestIds[requestId];
-        // Check if the request Id is invalid (non existent or delivered): previous and next request Ids are zero,
-        // and the zero's element previous request Id is not equal to the provided request Id
-        if (requestIds[0] == 0 && requestIds[1] == 0 && mapRequestIds[0][0] != requestId) {
-            revert RequestIdNotFound(requestId);
-        }
+    function _cleanRequestInfo(uint256 requestId) internal virtual {
+        // Get request Id from the request Ids map
+        uint256[2] memory requestIdLinks = mapRequestIds[requestId];
 
         // Re-link previous and next elements between themselves
-        mapRequestIds[requestIds[0]][1] = requestIds[1];
-        mapRequestIds[requestIds[1]][0] = requestIds[0];
+        mapRequestIds[requestIdLinks[0]][1] = requestIdLinks[1];
+        mapRequestIds[requestIdLinks[1]][0] = requestIdLinks[0];
 
         // Delete the delivered element from the map
         delete mapRequestIds[requestId];
     }
 
-    /// @dev Delivers a request.
+    /// @dev Prepares delivery of requests.
     /// @notice This function ultimately calls mech marketplace contract to finalize the delivery.
-    /// @param requestId Request id.
-    /// @param data Self-descriptive opaque data-blob.
-    function _deliver(uint256 requestId, bytes memory data) internal virtual returns (bytes memory requestData) {
-        // Get an requester to deliver request to
-        address requester = mapRequestAddresses[requestId];
-
-        // Get the mech delivery info from the mech marketplace
-        IMechMarketplace.MechDelivery memory mechDelivery =
-            IMechMarketplace(mechMarketplace).mapRequestIdDeliveries(requestId);
-
-        // Instantly return with empty data if the request has been delivered
-        // This allows not to fail batch requests transactions
-        if (mechDelivery.deliveryMech != address(0)) {
-            return "";
-        }
```

Comment:
No need to pull the actual request info from the MM and then cancel it. The delivered request statuses for each request Id is going to be pulled when calling MM delivery function. Then, the undelivered requests are cleaned up.